lib/sisimai/bite/email/aol.rb

```diff
@@ -0,0 +1,208 @@
+                end
+                v['recipient'] = cv[1]
+                recipients += 1
+
+              elsif cv = e.match(/\A[Aa]ction:[ ]*(.+)\z/)
+                # Action: failed
+                v['action'] = cv[1].downcase
+
+              elsif cv = e.match(/\A[Ss]tatus:[ ]*(\d[.]\d+[.]\d+)/)
+                # Status:5.2.0
+                v['status'] = cv[1]
+
+              elsif cv = e.match(/\A[Rr]emote-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Remote-MTA: DNS; mx.example.jp
+                v['rhost'] = cv[1].downcase
+
+              else
+                # Get error message
+                if cv = e.match(/\A[Dd]iagnostic-[Cc]ode:[ ]*(.+?);[ ]*(.+)\z/)
+                  # Diagnostic-Code: SMTP; 550 5.1.1 <userunknown@example.jp>... User Unknown
+                  v['spec'] = cv[1].upcase
+                  v['diagnosis'] = cv[2]
+
+                elsif p =~ /\A[Dd]iagnostic-[Cc]ode:[ ]*/ && cv = e.match(/\A[ \t]+(.+)\z/)
+                  # Continued line of the value of Diagnostic-Code header
+                  v['diagnosis'] ||= ''
+                  v['diagnosis']  += ' ' + cv[1]
+                  havepassed[-1] = 'Diagnostic-Code: ' + e
+                end
+              end
+
+            else
+              # Content-Type: message/delivery-status
+              # Content-Transfer-Encoding: 7bit
+              #
+              # Reporting-MTA: dns; omr-m5.mx.aol.com
+              # X-Outbound-Mail-Relay-Queue-ID: CCBA43800007F
+              # X-Outbound-Mail-Relay-Sender: rfc822; shironeko@aol.example.jp
+              # Arrival-Date: Fri, 21 Nov 2014 17:14:34 -0500 (EST)
+              if cv = e.match(/\A[Rr]eporting-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Reporting-MTA: dns; mx.example.jp
+                next if connheader['lhost'].size > 0
+                connheader['lhost'] = cv[1].downcase
+                connvalues += 1
+
+              elsif cv = e.match(/\A[Aa]rrival-[Dd]ate:[ ]*(.+)\z/)
+                # Arrival-Date: Wed, 29 Apr 2009 16:03:18 +0900
+                next if connheader['date'].size > 0
+                connheader['date'] = cv[1]
+                connvalues += 1
+              end
+
+            end
+          end
+        end
+
+        return nil if recipients.zero?
+        require 'sisimai/string'
+        require 'sisimai/smtp/status'
+
+        dscontents.map do |e|
+          # Set default values if each value is empty.
+          connheader.each_key { |a| e[a] ||= connheader[a] || '' }
+
+          e['agent']     = self.smtpagent
+          e['diagnosis'] = e['diagnosis'].gsub(/\\n/, ' ')
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          ReFailure.each_key do |r|
+            # Verify each regular expression of session errors
+            next unless e['diagnosis'] =~ ReFailure[r]
+            e['reason'] = r.to_s
+            break
+          end
+
+          if e['status'].empty? || e['status'] =~ /\A\d[.]0[.]0\z/
```

Comment:
__[rubocop]__
* Style/Next: Use `next` to skip iteration. [:ref](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)

<!-- Comment by [SideCI](https://sideci.com) -->